librz/util/zip.c

```diff
@@ -24,31 +24,33 @@ static const char *gzerr(int n) {
 }
 
 /**
- * @brief inflate zlib compressed or gzipped, automatically accepts either the zlib or gzip format, and use MAX_WBITS as the window size logarithm.
- * @see rz_inflatew()
+ * \brief inflate zlib compressed or gzipped, automatically accepts either the zlib or gzip format, and use MAX_WBITS as the window size logarithm.
+ * \see rz_inflatew()
  */
-RZ_API ut8 *rz_inflate(const ut8 *src, int srcLen, int *srcConsumed, int *dstLen) {
+RZ_API ut8 *rz_inflate(RZ_NONNULL const ut8 *src, int srcLen, int *srcConsumed, int *dstLen) {
 	return rz_inflatew(src, srcLen, srcConsumed, dstLen, MAX_WBITS + 32);
 }
 
 /**
- * @brief inflate zlib compressed or gzipped. The input must be a raw stream with no header or trailer.
- * @see rz_inflatew()
+ * \brief inflate zlib compressed or gzipped. The input must be a raw stream with no header or trailer.
+ * \see rz_inflatew()
  */
-RZ_API ut8 *rz_inflate_ignore_header(const ut8 *src, int srcLen, int *srcConsumed, int *dstLen) {
+RZ_API ut8 *rz_inflate_ignore_header(RZ_NONNULL const ut8 *src, int srcLen, int *srcConsumed, int *dstLen) {
 	return rz_inflatew(src, srcLen, srcConsumed, dstLen, -MAX_WBITS);
```

Comment:
Also `rz_return_val_if_fail()`

Reply:
But wouldn't that be redundant, since the non-null check is present in `rz_inflatew(...)`?

Reply:
These checks appear only in certain "debug" builds, during the release, packaging, etc they evaluate to nothing. See https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#c

Reply:
Alright, makes sense. I'll add the checks to make it consistent.